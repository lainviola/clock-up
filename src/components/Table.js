import styled, { css } from 'styled-components'
import { borderColor } from '../commonCSS'

const tableBorder = css`
  border: 1px solid ${borderColor};
  border-collapse: collapse;
`

const tablePadding = css`
  padding: 10px 8px;
`

export const Table = styled.table`
  margin: 0 auto;
  padding: 10px;
  ${tableBorder};
`

export const Th = styled.th`
  ${tableBorder};
  ${tablePadding};
`

export const Td = styled.td`
  ${tableBorder};
  ${tablePadding};
`

export const Tbody = styled.tbody`
  tr:nth-child(even) > Td {
    background-color: #fafafa;
  }
`
