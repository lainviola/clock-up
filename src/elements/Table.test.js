import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Table, Th, Tbody, Td } from './Table'

describe('<Table />', () => {
  it('should match to spapshot', () => {
    const wrapper = shallow(<Table />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('<Th />', () => {
  it('should match to spapshot', () => {
    const wrapper = shallow(<Th />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('<Tbody />', () => {
  it('should match to spapshot', () => {
    const wrapper = shallow(<Tbody />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('<Td />', () => {
  it('should match to spapshot', () => {
    const wrapper = shallow(<Td />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
