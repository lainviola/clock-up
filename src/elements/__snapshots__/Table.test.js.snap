// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match to spapshot 1`] = `
<table
  className="Table-evpfJW ekcGgi"
/>
`;

exports[`<Tbody /> should match to spapshot 1`] = `
<tbody
  className="Table__Tbody-coKAkO dHtLHf"
/>
`;

exports[`<Td /> should match to spapshot 1`] = `
<td
  className="Table__Td-rXNav LRKLc"
/>
`;

exports[`<Th /> should match to spapshot 1`] = `
<th
  className="Table__Th-eisASl fdsIwX"
/>
`;
